Fix to image captions extending beyond image width

mayagbarnes opened the pull request:
**Problem:**
    The StyledImageContainer is adjusting to the caption length, extending past the width of the image

**Fix:**
 * Added styling to StyledCaption (width, wordWrap, & padding) to ensure captions stay within image width and are readable (even at smaller dimensions)
 * Added test to validate caption width is the same as the image width
 

**Before:**
![image](https://user-images.githubusercontent.com/63436329/134090650-551ec4ed-e7ea-4a65-a0f1-8498cfde71f7.png)

**After:**
<img width="580" alt="Screen Shot 2021-09-20 at 4 46 13 PM" src="https://user-images.githubusercontent.com/63436329/134091340-01920c40-cdc6-48d0-a500-2565a393528b.png">

close #3530

mayagbarnes committed:
Add options for image caption bug fix

mayagbarnes committed:
Add spacing options

mayagbarnes committed:
Update styling to caption

mayagbarnes committed:
Add test to cover caption width

mayagbarnes committed:
Merge branch 'streamlit:develop' into image-caption

vdonato commented on frontend/src/components/elements/ImageList/ImageList.test.tsx:
```diff
@@ -92,6 +92,20 @@ describe("ImageList Element", () => {
   })
 
+  it("should render explicit width for each caption", () => {
+    const props = {
+      ...getProps({
+        width: 300,
+      }),
+      width: 1,
+    }
```
nit: I think the way `getProps` is written lets you equivalently write
```javascript
const props = getProps({ width: 1 })
```

vdonato reviewed (approved):
LGTM!

I was initially a bit confused as to why snapshot tests didn't fail and have to be updated since captions now look different if the text wraps, but it looks like we don't actually have a test that takes snapshots of both the image and its caption.

Would recommend adding an additional snapshot (I think just getting one for the first image+caption in the test would be fine) in `st_image.spec.js`

mayagbarnes committed:
Update te2e to check for caption width

mayagbarnes committed:
Add screenshot test for img + caption

mayagbarnes committed:
Merge branch 'image-caption' of https://github.com/mayagbarnes/streamlit into image-caption

mayagbarnes commented:
**Post-Review Updates:**
* Added snapshot test to st_image.spec (covering both img & caption) and updated existing caption test to include check for width
* Fixed instances of unnecessary getProps destructuring in Imagelist.test
